### ISSUE TITLE
# symbol parsing

### ISSUE THREAD

**Issue by plankter** [jmp75/rdotnet#14]
Since release of version 1.6 "#" symbols are not parsed. For instance, RManager.Engine.Evaluate("print("#something") does not print anything. It creates a lot of problems, since it is not possible to use color codes in ggplot2 anymore (i.e. "#E69F00"). Also knitr stop working because it cannot parse headers in the format "### Header text".

It seems the problem occurs due to changes in commit https://github.com/jmp75/rdotnet/commit/838720267f24ad1c720abc43349bfa4f8ffa5125:

<pre>
private static IEnumerable<string> Segment(string line)
         {
-            var segments = line.Split(';');
+            // Fix for
+            // https://rdotnet.codeplex.com/workitem/165
+            string[] commentsSegments = line.Split('#');
+            string uncommentedStatements;
+            if (commentsSegments.Length == 0)
+                uncommentedStatements = string.Empty;
+            else
+                uncommentedStatements = commentsSegments[0];
+
+            var segments = uncommentedStatements.Split(';');
</pre>

**Comment by skyguy94** [comment]
Wow. Talk about unintended consequences. Are you ok with reverting to an older version in nuget for now?

**Comment by plankter** [comment]
Yes, i just rolled back to 1.5.22. Thank you in advance for fixing this issue :)

**Comment by jmp75** [comment]
Yikes, I went too far in responding fast to an issue. Backfired; live and learn. I see the kind of unit tests we need. If you can think of further cases, report them. I vaguely dread a can of worm of nested text parsing for things like evaluate an expression that itself is an expression to evaluate.

**Comment by jmp75** [comment]
I have a candidate fix for this currently on the testing branch https://github.com/jmp75/rdotnet/commit/4306481cfabd213a507a3134b17f5f04fee6d8c3. Probably not the best solution for parsing, but this passes some non-trivial corner case tests with '#' and double quotes and quotes in R strings. Happy to receive feedback before an updated release.

**Comment by skyguy94** [comment]
I'll give it a try today

**Comment by plankter** [comment]
Unfortunately, in version 1.6.4 the parsing behavior is still not as it was in 1.5.22. For instance, knitr does not recognize markdown commands and there are no line separation as well in rmarkdown. Is it possible to have an option in the future to enable the old parsing mechanics instead of new one?

Thanks.
